modules/claims_api/spec/lib/claims_api/error/error_handler_spec.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'claims_api/special_issue_mappers/bgs'
+require 'token_validation/v2/client'
+require 'claims_api/error/error_handler'
+# require 'spec_helper'
+
+# describe ApplicationController, type: :controller do
+#   let(:scopes) { %w[claim.write] }
+#   controller do
+#     def test_invalid_token
+#       raise ::Common::Exceptions::TokenValidationError
+#     end
+#     skip_before_action :authenticate
+#   end
+
+#   before :each do
+#     routes.draw do
+#       get 'test_invalid_token' => 'anonymous#test_invalid_token'
+#     end
+#   end
+
+#   it 'catches an invalid token' do
+#     with_okta_user(scopes) do |auth_header|
+#       p auth_header
+#       request.headers.merge!(auth_header) # https://github.com/rspec/rspec-rails/issues/1655
+#       p request.headers
+#       # byebug
+#       get :test_invalid_token
+#       puts response.body
+#       expect(response).to have_http_status(401)
+#       parsed_body = JSON.parse(response.body)
+#       expect(parsed_body['detail']).to eq('Invalid token.')
+#     end
+#   end
+# end
+
+describe ApplicationController, type: :request do
+  let(:veteran_id) { '1013062086V794840' }
+  let(:scopes) { %w[claim.write] }
+  let(:get_poa_path) { "/services/benefits/v2/veterans/#{veteran_id}/power-of-attorney" }
+
+  it 'catches an invalid token' do
+    with_okta_user(scopes) do |auth_header|
+      allow(JWT).to receive(:decode).and_raise(JWT::DecodeError)
+
+      get get_poa_path, headers: auth_header
+
+      parsed_body = JSON.parse(response.body)
+      expect(parsed_body['errors'][0]['detail']).to eq('Invalid token.')
+    end
+  end
+
+  it 'catches an expired token' do
+    with_okta_user(scopes) do |auth_header|
+      # allow(JWT).to receive(:decode).and_raise(JWT::ExpiredSignature)
+      i = 0
+      allow(JWT).to receive(:decode) do
+        # raise JWT::ExpiredSignature
+        raise(JWT::ExpiredSignature) if i > 2
+
+        i += 1
+        okta_jwt(scopes)
+      end
+
+      get get_poa_path, headers: auth_header
+
+      parsed_body = JSON.parse(response.body)
+      expect(parsed_body['errors'][0]['detail']).to eq('Invalid token.')
+    end
+  end
```

Comment:
Let's remove the commented code from within this test.

I don't really love this test because it seems unnecessarily complex.
But we can revisit it once we get to focus on V2 again.

Reply:
[I was able to simplify this a lot](https://github.com/department-of-veterans-affairs/vets-api/pull/9444/commits/5dfb11e61f79393ab91ae30fe78a827e995b09fe) by just stubbing those 3 calls explicitly. It now fails where we want it to.